src/extension.ts

```diff
@@ -852,6 +865,195 @@ export async function activate(context: vscode.ExtensionContext) {
+      );
+      if (!shallDelete || shallDelete.title === "Cancel") {
+        return;
+      }
+      vscode.window.withProgress(
+        {
+          title: "Deleting node from your rainmaker account",
+          location: vscode.ProgressLocation.Notification,
+        },
+        async () => {
+          try {
+            await RainmakerAPIClient.deleteNode(item.id);
+            rainMakerTreeDataProvider.refresh();
+          } catch (error) {
+            Logger.errorNotify(
+              "Failed to delete node, maybe the node is already marked for delete, please try again after sometime",
+              error
+            );
+          }
+        }
+      );
+    }
+  );
+  registerIDFCommand("esp.rainmaker.backend.add_node", async () => {
+    Logger.infoNotify(
+      "Coming Soon!! until then you can add nodes using mobile app"
+    );
+  });
+  registerIDFCommand(
+    "esp.rainmaker.backend.update_node_param",
+    async (item: RMakerItem) => {
+      if (!item) {
+        return;
+      }
+      const idPayload = item.id.split("::");
+      const params = item.getMeta<RainmakerDeviceParamStructure>();
+
+      if (params.properties.indexOf("write") === -1) {
+        return Logger.infoNotify("Readonly Property");
+      }
+
+      let newParamValue = await vscode.window.showInputBox({
+        ignoreFocusOut: true,
+        placeHolder: "param value",
+        value: item.description.toString(),
+        prompt: "Enter the new param value",
+        validateInput: (value: string): string => {
+          return validateInputForRainmakerDeviceParam(value, params.data_type);
+        },
+      });
+
+      if (!newParamValue) {
+        return;
+      }
+
+      newParamValue = convertTo(params.data_type, newParamValue);
+
+      vscode.window.withProgress(
+        {
+          title: "Syncing params, please wait",
+          location: vscode.ProgressLocation.Notification,
+        },
+        async () => {
+          try {
+            const nodeID = idPayload[0];
+            const deviceName = idPayload[1];
+            await RainmakerAPIClient.updateNodeParam(
+              nodeID,
+              deviceName,
+              params.name,
+              newParamValue
+            );
+            await rainMakerTreeDataProvider.refresh();
+            Logger.infoNotify("Sent the param update request to cloud");
+          } catch (error) {
+            Logger.errorNotify(
+              "Failed to update the param, please try once more",
+              error
+            );
+          }
+        }
+      );
+    }
+  );
+  vscode.window.registerUriHandler({
+    handleUri: (uri: vscode.Uri) => {
+      Logger.info(`vscode:// URI invoke received, ${uri.toString()}`);
+    },
+  });
+}
+
+function validateInputForRainmakerDeviceParam(
+  value: string,
+  type: string
+): string {
+  if (type === "bool" && !value.match(/^(true|false)$/i)) {
+    return "Enter valid boolean value (true or false)";
+  }
```

Comment:
Wouldn't it be better to use `vscode.window.showQuickPick` with true/false for bool type here?

Beside this, everything else LGTM!